### ISSUE TITLE
Feature Request: Support Secrets Manager Secret Resource Policies

### ISSUE THREAD

**Issue by bflad** [terraform-providers/terraform-provider-aws#4997]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS Secrets Manager is adding resource policies to secrets.

Prerequisite: AWS Go SDK v1.14.14 (#4996)

### New or Affected Resource(s)

* `aws_secretsmanager_secret`

### Potential Terraform Configuration

```hcl
# Example implementation, may change during development
resource "aws_secretsmanager_secret" "example" {
  # ... other configuration ...
  policy = "{...}"
}
```

### References

* https://github.com/aws/aws-sdk-go/releases/tag/v1.14.14

**Comment by tsiq-tek** [comment]
Submitted #5290 that implements the proposed syntax.

**Comment by ts-tek** [comment]
Submitted #5290  that implements the proposed syntax.

**Comment by bflad** [comment]
The above pull request was merged and the new `policy` argument for the `aws_secretsmanager_secret` resource will be release in version 1.29.0 of the AWS provider, likely middle of this week.

**Comment by bflad** [comment]
This has been released in [version 1.29.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1290-july-26-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
